queries/coingecko/useCoinGeckoCoinListQuery.ts

```diff
@@ -0,0 +1,35 @@
+import axios from 'axios';
+import { QueryConfig, useQuery } from 'react-query';
+import keyBy from 'lodash/keyBy';
+
+import QUERY_KEYS from 'constants/queryKeys';
+import { CurrencyKey } from 'constants/currency';
+
+import { CG_BASE_API_URL } from './constants';
+
+type CoinListItem = {
+	id: string;
+	symbol: CurrencyKey;
+	name: string;
+};
+
+type CoinListMap = Record<CurrencyKey, CoinListItem>;
+
+const useCoinGeckoCoinListQuery = (options?: QueryConfig<CoinListMap>) => {
```

Comment:
I ended up not using this query in favour of `useCoinGeckoTokenPricesQuery` which accepts token addresses... However, since I already created it, I thought maybe in the future it will be useful.